src/compile/Component.ts

```diff
@@ -987,11 +987,11 @@ export default class Component {
 						const { name } = flatten_reference(node);
 						const owner = scope.find_owner(name);
 
-						if (name[0] === '$' && !owner) {
+						if (node.type === 'Identifier' && injected_reactive_declaration_vars.has(name)) {
```

Comment:
Is there a particular reason you're keeping these two in separate branches, rather than just using an 'or'?

Reply:
Readability mostly. I thought the line would get too long that way, and I couldn't find any if statements like ...
```javascript
if (
  expression1 ||
  expression2
) {
  // ...
}
```

... in the codebase. Would you prefer both on the same line, or something like the example above?